augur/api/view/routes.py

```diff
@@ -48,163 +60,126 @@ def logo(brand=None):
-
-            if not user and register is None:
-                raise LoginException("Invalid login credentials")
+            user = User.get_by_login(db_session, username)
+            if not user:
+                raise LoginException("Invalid username or password")
             
-            # register a user
-            if register is not None:
-                if user:
-                    raise LoginException("User already exists")
-                
-                email = request.form.get('email')
-                first_name = request.form.get('first_name')
-                last_name = request.form.get('last_name')
-                admin = request.form.get('admin') or False
-
-                result = User.create_user(username, password, email, first_name, last_name, admin)
-                if not result[0]:
-                    raise LoginException("An error occurred registering your account")
-                else:
-                    user = User.get_user(db_session, username)
-                    flash(result[1]["status"])
-
-            # Log the user in if the password is valid
-            if user.validate(password) and login_user(user, remember = remember):
-                flash(f"Welcome, {username}!")
-                if "login_next" in session:
-                    return redirect(session.pop("login_next"))
-                return redirect(url_for('root'))
-            else:
-                print("Invalid login")
-                raise LoginException("Invalid login credentials")
+            if not user.check_password(password):
+                raise LoginException("Invalid username or password")
+            
+            login_user(user)
+            flash("Login successful", "success")
+            return redirect(request.args.get('next', url_for('user_settings')))
         except LoginException as e:
-            flash(str(e))
-    return render_module('login', title="Login")
+            flash(str(e), "error")
+            return render_template("login.j2")
+        
+    return render_template("login.j2")
 
 """ ----------------------------------------------------------------
-logout:
-    Under development
+    This route ends a user's session.
 """
 @app.route('/account/logout')
 @login_required
 def user_logout():
     logout_user()
-    flash("You have been logged out")
-    return redirect(url_for('root'))
+    flash("You have been logged out", "success")
+    return redirect(url_for('repo_table_view'))
 
 """ ----------------------------------------------------------------
-default:
-table:
-    This route performs external authorization for a user
+    This route authorizes a 3rd party application to access a
+    user's account via the API.
 """
 @app.route('/user/authorize')
 @login_required
 def authorize_user():
-    client_id = request.args.get("client_id")
-    state = request.args.get("state")
-    response_type = request.args.get("response_type")
-
-    if not client_id or response_type != "code":
```

Comment:
Checking the `response_type` is part of the Oauth standard, and cannot be omitted.